osu.Game/Screens/OnlinePlay/Lounge/DrawableLoungeRoom.cs

```diff
@@ -178,46 +179,67 @@ public class PasswordEntryPopover : OsuPopover
         {
             private readonly Room room;
 
-            public Action<Room, string> JoinRequested;
+            public Action<Room, string, Action<Room>, Action<string>> JoinRequested;
```

Comment:
This delegate signature is pretty egregious. Would be good to not have nested `Action`s if at all possible. Maybe the lounge room should resolve the lounge and call `Join()` directly or something of the sort.

Reply:
I'm thinking I can just pass down the already-resolved `LoungeSubScreen` from `DrawableLoungeRoom` on construction. I'll try it out and see if it works and update the PR if it does.